src/main/scala/com/scality/clueso/tools/MetadataTableCompactorTool.scala

```diff
@@ -0,0 +1,44 @@
+package com.scality.clueso.tools
+
+import java.io.File
+
+import com.scality.clueso.compact.TableFilesCompactor
+import com.scality.clueso.{CluesoConfig, SparkUtils}
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.LazyLogging
+import org.rogach.scallop._
+
+object MetadataTableCompactorTool extends LazyLogging {
+
+  class ToolConf(arguments: Seq[String]) extends ScallopConf(arguments) {
+    val applicationConfFile = trailArg[String](required = true, descr = "application configuration file")
+    val numPartitions = trailArg[Int](required = true)
+    val bucket = trailArg[String](required = false)
+    verify()
+  }
+
+  def main(args: Array[String]): Unit = {
+    runTool(new ToolConf(args))
+  }
+
+  def runTool(toolConfig: ToolConf) = {
+    val parsedConfig = ConfigFactory.parseFile(new File(toolConfig.applicationConfFile.getOrElse("")))
+    val _config = ConfigFactory.load(parsedConfig)
+
+    val config = new CluesoConfig(_config)
+
+    val spark = SparkUtils.buildSparkSession(config)
+      .master("local[*]")
+      .appName("Table Files Compactor")
+      .getOrCreate()
+
+
+    val merger = new TableFilesCompactor(spark, config)
+
+    if (toolConfig.bucket.supplied) {
+      merger.compactLandingPartition("bucket", toolConfig.bucket.apply(), toolConfig.numPartitions.apply(), false)
+    } else {
+      merger.compact(toolConfig.numPartitions.apply(), false)
```

Comment:
don't you have to call the function with null, null for the bucket and bucketName arguments?

Reply:
there's compactLandingPartition  which compacts a bucket partition (based on force flag and number of maxOpIndex subpartitions) and there's compact() which goes through all the available buckets and calls compactLandingPartition on each one